misc/bstr.c

```diff
@@ -393,14 +393,15 @@ void bstr_xappend_vasprintf(void *talloc_ctx, bstr *s, const char *fmt,
     va_copy(copy, ap);
     size_t avail = talloc_get_size(s->start) - s->len;
     char *dest = s->start ? s->start + s->len : NULL;
-    char c;
-    if (avail < 1)
-        dest = &c;
-    size = vsnprintf(dest, MPMAX(avail, 1), fmt, copy);
+    assert(fmt);
+    size = vsnprintf(dest, avail, fmt, copy);
     va_end(copy);
 
-    if (size < 0)
-        abort();
+    if (size < 0) {
+        bstr_xappend(talloc_ctx, s, bstr0("format error: "));
+        bstr_xappend(talloc_ctx, s, bstr0(fmt));
```

Comment:
writing an error message to the destination string of sprintf fails sounds like a terrible idea

Reply:
Why?

Reply:
you mean vsnprintf only fails deterministically?

> why

garbage-in, garbage-out is not a good principle

Reply:
> garbage-in, garbage-out is not a good principle

Like I said, we should never pass dynamically generated format. All our formats should be nice and clean. The issue is with the arguments that fails on, specifically some unicode garbage, which may be injected by the user.

Reply:
I wanted easy way for this. It is mostly logging. But I can move this to the callers side, but this would still be very similar in practice, because it is deep in printing code anyway.

Reply:
updated